### DISCUSSION TITLE
how add custom labels to vc-measurement / drawing

### DISCUSSION THREAD

**Discussion by PetukhovArt** [zouyaoji/vue-cesium#608] [Q&A]
Hello !
i want to add label on drawing with string name of this drawing and position it in the center ( or some custom position)
how better to do it ?

**Comment by zouyaoji** [comment]
![image](https://github.com/user-attachments/assets/8e16d978-90de-4152-894b-98b87da9ad10)

**Reply by zouyaoji** [reply]
The measurement component can be modified after drawing. The drawing component needs to enable label display before making changes.

**Reply by PetukhovArt** [reply]
hi! i found the problem with updating showLabel state of each drawing.
case : 

1. fetch data and initialize drawings with options :  labelOpts.show =false / labelsOpts.show = false
2. then, try to change this state in renderDatas to true , and labels do not showing.

this working if :
1) after drawingsReady i initialize drawings with labelOpts.show / labelsOpts.show   status true
2) timeout 1000
3) set drawings  labelOpts.show / labelsOpts.show   status false

then i can change labels state and all is working.

how i can control labels state , without this?  or may be my solution is wrong and i need to edit renderDatas labels ..etc with **value[i].labels[i]** ? 

![image](https://github.com/user-attachments/assets/89fe0185-10f3-4af2-90c6-f4d4651a4e4a)
